src/AppBundle/Issues/IssueListener.php

```diff
@@ -58,14 +58,39 @@ public function handleCommentAddedEvent($issueNumber, $comment)
      * Adds a "Needs Review" label to new PRs.
      *
      * @param int $prNumber The number of the PR
+     * @param int $prTitle  The title of the PR
+     * @param int $prBody   The full text description of the PR
      *
      * @return string The new status
      */
-    public function handlePullRequestCreatedEvent($prNumber)
+    public function handlePullRequestCreatedEvent($prNumber, $prTitle, $prBody)
     {
+        $prLabels = array();
+
+        // new PRs always require review
         $newStatus = Status::NEEDS_REVIEW;
+        $prLabels[] = $newStatus;
+
+        // the PR title usually indicates the affected component
```

Comment:
> Imo, we should get a list of available labels for a repository and check all brackets as long as the value inside the brackets do not match a label

You are right ... but I prefer to leva that for the future. For now I've hardcoded the valid labels, which sounds awful ... but it's not that bad because they rarely change.